evo_prophet/autonolas/research.py

```diff
@@ -885,41 +887,40 @@ def process_in_batches(
     }
     session.headers.update(headers)
 
-    # Using ThreadPoolExecutor to execute requests in parallel
-    with ThreadPoolExecutor() as executor:
-        # Loop through the URLs in batch_size of size 'batch_size'
-        for i in range(0, len(urls), batch_size):
-            batch = urls[i : i + batch_size]
-
-            # Submit the batch of URLs for processing
-            futures = []
-            for url in batch:
-                try:
-                    # Submit a HEAD request to the url and check Content-Type
-                    head_future = executor.submit(
-                        session.head,
-                        url,
-                        headers=headers,
-                        timeout=timeout,
-                        allow_redirects=True,
-                    )
-                    head_response = head_future.result()
-                    if "text/html" not in head_response.headers.get("Content-Type", ""):
-                        continue
-                    else:
-                        # Submit a GET request to the url
-                        futures.append(
-                            (
-                                executor.submit(
-                                    session.get, url, headers=headers, timeout=timeout
-                                ),
-                                url,
-                            )
+    # Using THREADPOOL to execute requests in parallel
```

Comment:
How does this change affect memory usage?

Reply:
My hope was that if instead of re-creating threadpool again and again, we use one fixed with fixed number of threads (instead of using all available of the system), it would be easier for GC. But idk how much this actually helped.